interface/ceed-vector.c

```diff
@@ -493,6 +493,103 @@ int CeedVectorNorm(CeedVector vec, CeedNormType norm_type, CeedScalar *norm) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Compute y = alpha x + y
+
+  @param y[in/out]  target vector for sum
+  @param alpha[in]  scaling factor
+  @param x[in]      second vector, must be different than y
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedVectorAXPY(CeedVector y, CeedScalar alpha, CeedVector x) {
+  int ierr;
+  CeedScalar *y_array;
+  CeedScalar const *x_array;
+  CeedInt n_x, n_y;
+
+  if (y->AXPY)
+    return y->AXPY(y, alpha, x);
+
+  ierr = CeedVectorGetLength(y, &n_y); CeedChk(ierr);
+  ierr = CeedVectorGetLength(x, &n_x); CeedChk(ierr);
+  if (n_x != n_y)
+    // LCOV_EXCL_START
+    return CeedError(y->ceed, CEED_ERROR_UNSUPPORTED,
+                     "Cannot add vector of different lengths");
+  // LCOV_EXCL_STOP
```

Comment:
Seems like we should do this input validation, as well as checking that `x->ceed == y->ceed`, before calling `y->AXPY`.